samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -154,12 +154,13 @@ def __init__(
 
         self._debug_function = debug_function
 
-        # Note(xinhol): despite self._template_dict and self._function_provider are initialized as None
+        # Note(xinhol): despite self._template_dict, self._function_provider and self._stacks are initialized as None
         # they will be assigned with a non-None value in __enter__() and
         # it is only used in the context (after __enter__ is called)
         # so we can assume they are not Optional here
         self._template_dict: Dict = None  # type: ignore
         self._function_provider: SamFunctionProvider = None  # type: ignore
+        self._stacks: List[Stack] = None  # type: ignore
```

Comment:
I would prefer the `Stack` object to have a List[Stack] property and a getter method to get all stacks and just pass the root stack instead of passing them all

Reply:
Storing it in a tree like structure makes sense for readability. Since this is a late stage PR, refactoring now would change a lot of work done in several previous PRs. I would hold this suggestion for now and it is a good future improvement. What do you think?